### ISSUE TITLE
Chapter 22 Export task stuck at 0%

### ISSUE THREAD

**Issue by CoreyCCarter** [miguelgrinberg/microblog#348]
I am working through chapter 22 of the tutorial and so far everything is good. I do have one issue, however, with the export task. When I export the posts, I get the green bar at the top of the screen but it stays stuck at 0%, even after 500 seconds have passed. If I query app.db via sqlite3, it shows the export task with a progress of 0% as well. The status of the job is 'queued' when I monitor it via rediscli.  I am not sure where to go from here as I feel like everything I have in code is 100% correct. Anyone have the same problem? Any ideas on how to proceed?

**Comment by miguelgrinberg** [comment]
Are you running a RQ worker process? This is a separate process that picks up the tasks that are put in the queue and executes them. Without the worker the task will never progress.

**Comment by CoreyCCarter** [comment]
I forgot I needed to be running the RQ worker process.  Thank You. How do I go about setting the Flask-Mail variables in the context of the RQ worker, though? I thought that was all set in my .env file that is loaded from config.py

**Comment by miguelgrinberg** [comment]
The configuration is done in the .env file, as you describe. Have you tested sending emails before? The most likely cause of the error is that your configuration is wrong.

**Comment by CoreyCCarter** [comment]
The reset password emails work perfectly, I just cannot get the export posts email to work and everything points back to the smtp error.

**Comment by CoreyCCarter** [comment]
@miguelgrinberg Thank you for your help with this. I restarted the application and started the worker process and everything is working.  I did, however, notice that when a user has no posts and asks for them to be exported, it will send the email with an empty json file(empty as in it has no post data but it does have the structure for it) but the progress bar and job itself gets stuck because of the 100 * i // total_posts calculation will result in division by zero. It works perfectly where there are actual posts, though.
